src/customio.cpp

```diff
@@ -0,0 +1,242 @@
+        m_Key(key), m_Handler(handler) {}
+
+    AddOutputKey_t     m_Key;
+    AddOutputHandler_t m_Handler;
+};
+
+static void AddOutputCustom_Targetname(Z_CBaseEntity*                  pInstance,
+                                       CEntityInstance*                pActivator,
+                                       CEntityInstance*                pCaller,
+                                       const std::vector<std::string>& vecArgs)
+{
+    addresses::CEntityIdentity_SetEntityName(pInstance->m_pEntity, vecArgs[1].c_str());
+    Message("SetName %s to %d", vecArgs[1].c_str(), pInstance->GetHandle().GetEntryIndex());
+}
+
+static void AddOutputCustom_Origin(Z_CBaseEntity*                  pInstance,
+                                   CEntityInstance*                pActivator,
+                                   CEntityInstance*                pCaller,
+                                   const std::vector<std::string>& vecArgs)
+{
+    Vector origin(clamp(Q_atof(vecArgs[1].c_str()), -16384.f, 16384.f),
+                  clamp(Q_atof(vecArgs[2].c_str()), -16384.f, 16384.f),
+                  clamp(Q_atof(vecArgs[3].c_str()), -16384.f, 16384.f));
+    pInstance->Teleport(&origin, nullptr, nullptr);
+    Message("SetOrigin %f %f %f for %s", origin.x, origin.y, origin.z, pInstance->GetName());
+}
+
+static void AddOutputCustom_Angles(Z_CBaseEntity*                  pInstance,
+                                   CEntityInstance*                pActivator,
+                                   CEntityInstance*                pCaller,
+                                   const std::vector<std::string>& vecArgs)
+{
+    QAngle angles(clamp(Q_atof(vecArgs[0].c_str()), -360.f, 360.f),
+                  clamp(Q_atof(vecArgs[1].c_str()), -360.f, 360.f),
+                  clamp(Q_atof(vecArgs[2].c_str()), -360.f, 360.f));
+    pInstance->Teleport(nullptr, &angles, nullptr);
+    Message("SetAngles %f %f %f for %s", angles.x, angles.y, angles.z, pInstance->GetName());
+}
+
+static void AddOutputCustom_MaxHealth(Z_CBaseEntity*                  pInstance,
+                                      CEntityInstance*                pActivator,
+                                      CEntityInstance*                pCaller,
+                                      const std::vector<std::string>& vecArgs)
+{
+    pInstance->m_iMaxHealth(clamp(Q_atoi(vecArgs[1].c_str()), 1, 99999999));
+    const int m_iMaxHealth = pInstance->m_iMaxHealth;
+    Message("SetMaxHealth %d for %s", m_iMaxHealth, pInstance->GetName());
+}
+
+static void AddOutputCustom_Health(Z_CBaseEntity*                  pInstance,
+                                   CEntityInstance*                pActivator,
+                                   CEntityInstance*                pCaller,
+                                   const std::vector<std::string>& vecArgs)
+{
+    const auto value = clamp(Q_atoi(vecArgs[1].c_str()), 1, 99999999);
```

Comment:
IIRC some maps out there can set health to 0 or -1, not to mention there's already the `SetHealth` input.

Reply:
add health 0 to pawn will cause some bug.
I think we also can proxy to ``SetHealth``